Add breaking change "Material Chip button semantics"

nscobie opened the pull request:
This is a pretty minor breaking change which shouldn't affect many people, but it's my first time writing a breaking change doc so please let me know what I could improve.

**Note: currently unsure about timeline, will update that soon.**

**Relevant PRs:**
Original: flutter/flutter#60141
Revert: flutter/flutter#60645
Re-land: flutter/flutter#61048

gaaclarke commented:
Technical details LGTM, I'll defer to @Sfshaza clarity and communications feedback =)

gaaclarke commented:
@kwalrath can you give it a look please since shams is out?

gaaclarke commented on src/docs/release/breaking-changes/material-chip-button-semantics.md:
```diff
@@ -0,0 +1,223 @@
+* [Revert "Tweaking Material Chip a11y semantics to match buttons (#60141)"][]
+* [Re-land "Tweaking Material Chip a11y semantics to match buttons (#60141)"][]
+
+[`Semantics`]: {{site.api}}/flutter/widgets/Semantics-class.html
+[`SemanticsProperties.button`]:
+{{site.api}}/flutter/semantics/SemanticsProperties/button.html
+[`SemanticsProperties.enabled`]:
+{{site.api}}/flutter/semantics/SemanticsProperties/enabled.html [`Chip`]:
+{{site.api}}/flutter/material/Chip-class.html [`ActionChip`]:
+{{site.api}}/flutter/material/ActionChip-class.html [`ChoiceChip`]:
+{{site.api}}/flutter/material/ChoiceChip-class.html [`FilterChip`]:
+{{site.api}}/flutter/material/FilterChip-class.html [`InputChip`]:
+{{site.api}}/flutter/material/InputChip-class.html
```
I think this may have gotten mixed up by newlines

nscobie replied:
Good catch, fixed now.

kwalrath commented on src/docs/release/breaking-changes/material-chip-button-semantics.md:
```diff
@@ -0,0 +1,177 @@
+
+## Migration guide
+
+**Most developers likely won't need to perform any migration.** This change will only affect developers who worked around the issue of Material Chips missing `button` semantics by wrapping the widget given to the `label` field of a `Chip` with a `Semantics` widget marked as `button: true`. **In this case, the inner and outer `button` semantics will conflict, resulting in the tappable area of the  ...
```
Address the developer as "you". (https://developers.google.com/style/person)

kwalrath commented:
One last thing... when in doubt, leave info out. The less text there is, the easier it is to find what you need.

You could just comment it out if you're unsure about whether it might be useful.

nscobie commented on src/docs/release/breaking-changes/material-chip-button-semantics.md:
```diff
@@ -0,0 +1,177 @@
+
+In `package:flutter/material.dart`, the `build(...)` method of the `_RawChipState` class (used by all chips) returns the resulting built chip widget wrapped in a [Semantics](https://api.flutter.dev/flutter/widgets/Semantics-class.html) widget.
+
+This method looked akin to the following:
```
Very good points! I've trimmed it down quite a bit, just focusing on the higher-level points developers might care about.

 I think I got confused by the phrasing in the [template doc](https://github.com/flutter/website/blob/master/src/docs/release/breaking-changes/template.md#description-of-change). It totally makes sense in hindsight, but maybe "SDK" -> "API"?

kwalrath replied:
I think you're right about that. Would you mind creating a PR to fix that?

nscobie replied:
Will do.

nscobie commented:
Thanks @kwalrath, that was all really great feedback! I've been perusing the documentation style guide you linked, and that's very helpful as well.

Apologies for the rush, I am open to either making further edits or letting you edit as you see fit, however you'd like to handle it. I took a crack at addressing your feedback so far, but let me know if I missed anything. :)

kwalrath commented on src/docs/release/breaking-changes/material-chip-button-semantics.md:
```diff
@@ -0,0 +1,156 @@
+* [`ChoiceChip`][]
+* [`FilterChip`][]
+* [`InputChip`][]
+* [Material Buttons][]
```
Should this link point to {{site.material}}, like the next one does? (I was surprised that two links that looked similar didn't go to similar places.)

(If this should go to site.material but the earlier usage of [Material Buttons] shouldn't, then use a new link name for the earlier one.)

nscobie replied:
Good catch! Updated it to `{{site.material}}`.

kwalrath commented on src/docs/release/breaking-changes/material-chip-button-semantics.md:
```diff
@@ -1,126 +1,60 @@
+you worked around the issue of Material Chips missing `button` semantics by
+wrapping the widget given to the `label` field of a `Chip` with a `Semantics`
+widget marked as `button: true`. **In this case, the inner and outer `button`
+semantics will conflict, resulting in the tappable area of the button shrinking
```
```suggestion
semantics conflict, resulting in the tappable area of the button shrinking
```

kwalrath commented on src/docs/release/breaking-changes/material-chip-button-semantics.md:
```diff
@@ -1,126 +1,60 @@
+wrapping the widget given to the `label` field of a `Chip` with a `Semantics`
+widget marked as `button: true`. **In this case, the inner and outer `button`
+semantics will conflict, resulting in the tappable area of the button shrinking
+down to the size of the label after this change is introduced.** This issue can
+be fixed by deleting that `Semantics` widget and replacing it with its child, or
+simply removing the `button: true` property if other semantic properties still
```
```suggestion
down to the size of the label after this change is introduced.** Fix this issue
either by deleting that `Semantics` widget and replacing it with its child,
or by removing the `button: true` property if other semantic properties still
```

kwalrath commented on src/docs/release/breaking-changes/material-chip-button-semantics.md:
```diff
@@ -1,126 +1,60 @@
-widget wrapped in a
-[Semantics](https://api.flutter.dev/flutter/widgets/Semantics-class.html)
-widget.
+The outermost [`Semantics`][] widget which wraps all Chip classes to describe
```
When you see "which" without a comma, as a rule either insert a comma or replace it with "that". https://developers.google.com/style/commas#setting-off-other-kinds-of-clauses 

```suggestion
The outermost [`Semantics`][] widget that wraps all Chip classes to describe
```

kwalrath commented:
Staged: https://developers.google.com/style/commas#setting-off-other-kinds-of-clauses
